web/src/components/Button.tsx

```diff
@@ -1,54 +1,150 @@
+  type: 'primary' | 'secondary' | 'secondary-elevated' | 'tertiary' | 'link';
   href?: string;
-  className?: string;
+  backgroundClasses?: string;
+  forgroundClasses?: string;
+  focusOutlineColor?: string;
   onClick?: () => void;
 }
 
 export function Button({
   icon,
   children,
   disabled,
-  background,
-  textColor,
   href,
-  className,
+  backgroundClasses, // backgroundColor, borderColor, margin, etc.
+  forgroundClasses, // textColor, etc.
+  focusOutlineColor,
+  size,
+  type,
   onClick,
 }: ButtonProps) {
   const renderAsLink = Boolean(href);
   const As = renderAsLink ? 'a' : 'button';
-  const type = renderAsLink ? undefined : 'button';
+  const componentType = renderAsLink ? undefined : 'button';
   const isIconOnly = !children && Boolean(icon);
 
   return (
-    <As
+    <div
       className={twMerge(
-        `mx-2 my-3 flex min-h-[45px] w-fit items-center justify-center gap-x-2 rounded-full bg-white px-2 py-3 text-md font-bold shadow-[0px_0px_13px_rgb(0_0_0/12%)] transition duration-200 hover:shadow-[0px_0px_23px_rgb(0_0_0/20%)] dark:bg-gray-600 dark:hover:shadow-[0px_0px_23px_rgb(0_0_0/50%)]`,
-        !isIconOnly && BUTTON_MIN_WIDTH_CLASS,
-        `${disabled ? 'opacity-60 hover:shadow-[0px_0px_13px_rgb(0_0_0/12%)]' : ''}`,
-        className
+        `items-center justify-center rounded-full ${getBackground(type, disabled)}`,
+        backgroundClasses
       )}
-      translate="no"
-      disabled={disabled}
-      style={{ color: textColor, background: background }}
-      href={href}
-      type={type}
-      onClick={onClick}
     >
-      {icon}
-      {children}
-    </As>
+      <As
+        className={twMerge(
+          `flex w-full flex-row items-center justify-center rounded-full text-sm font-semibold disabled:text-neutral-400
+        disabled:hover:bg-inherit disabled:dark:text-gray-500 ${getSize(
+          size,
+          type,
+          isIconOnly
+        )}
+        ${getFocused(focusOutlineColor)} ${getForeground(type)} ${getHover(type)}`,
+          forgroundClasses
+        )}
+        disabled={disabled}
+        href={href}
+        type={componentType}
+        onClick={onClick}
+      >
+        {icon}
+        {children}
+      </As>
+    </div>
   );
 }
+
+function getHover(type: string) {
+  switch (type) {
+    case 'primary': {
+      return 'hover:bg-black/20';
+    }
+    default: {
+      return 'hover:bg-neutral-400/10';
+    }
+  }
+}
+
+function getBackground(type: string, disabled: boolean | undefined) {
+  switch (type) {
+    case 'primary': {
+      if (disabled) {
+        return 'bg-zinc-50 dark:bg-gray-800 border border-neutral-200 dark:border-gray-700';
+      }
+      return 'bg-brand-green';
+    }
+    case 'secondary': {
+      return 'border dark:border-gray-700 border-neutral-200 bg-inherit';
+    }
+    case 'secondary-elevated': {
+      return 'border dark:border-gray-700 border-neutral-200 bg-neutral-100 dark:bg-gray-800';
+    }
```

Comment:
Will we actually be using the `secondary` type in the new design changes? Otherwise I'd prefer to rename `secondary-elevated` to just `secondary`, as I find the "elevated" name quite confusing. In my mind, elevation means it has shadows and is "lifted" above something else, but here it just has a background color 🤔 

On another note, I think it would be better to call it "[ghost](https://uxplanet.org/ghost-buttons-in-ux-design-4cf3717334f8)" or "[outline](https://m3.material.io/components/buttons/guidelines#3742b09f-c224-43e0-a83e-541bd29d0f05)" to follow common naming practices. This makes it a bit easier to know what it is, as "secondary" does not say anything about how it will look :)

cc @Alportan as this is also the case in Figma

Reply:
Hmmm, I went with `primary`, `secondary`, `tertiary`, `link` approach as they are more semantically showing hierarchy if that makes sense. 

I removed the `secondary-elevated` in the meantime for simplicity but maybe Silke implemented while I was also working on the components 🙈. That's on me, I owe you an apology @silkeholmebonnen .

Reply:
We should still do the naming convention survey and get a common understanding of common naming conventions. Ideally we should roll it out in our Contrib Slack as well to get a more complete overview. 🤔

Reply:
Hmm yeah, I guess that makes sense - I'm fine with this for now, we can always bring the topic back up another time if it becomes too confusing :)